pkg/writer.go

```diff
@@ -107,6 +142,108 @@ func (s *writerImpl) Close() (err error) {
 	return
 }
 
+type encodeResult struct {
+	buf   []byte
+	entry seekTableEntry
+}
+
+func (s *writerImpl) writeManyEncoder(ctx context.Context, ch chan<- encodeResult, frame []byte) func() error {
+	return func() error {
+		dst, entry, err := s.encodeOne(frame)
+		if err != nil {
+			return err
+		}
+
+		select {
+		case <-ctx.Done():
+		// Fulfill our promise
+		case ch <- encodeResult{dst, entry}:
+			close(ch)
+		}
+
+		return nil
+	}
+}
+
+func (s *writerImpl) writeManyReader(ctx context.Context, frameSource FrameSource, g *errgroup.Group, queue chan<- chan encodeResult) func() error {
+	return func() error {
+		for {
```

Comment:
I'm not sure we really care about closing channels tbh?[ It's perfectly safe to allow an open channel to be garbage collected](https://groups.google.com/g/golang-nuts/c/pZwdYRGxCIk/m/qpbHxRRPJdUJ), it will be auto-closed once nothing references it anymore.

The only consideration is whether anybody stuck waiting on the channel, so that GC is impossible. In our case, I'm pretty sure the answer is no due to how we use select.

Reply:
yeah, i'm mostly afraid about the consumer stall.  That said, I do see your point, in the current version, if this goroutine exits with an error, the consumer would select on the ctx.Done.